src/main/java/com/singlestore/fivetran/destination/JDBCUtil.java

```diff
@@ -43,7 +43,20 @@ static Connection createConnection(SingleStoreConfiguration conf) throws Excepti
         }
 
         String url = String.format("jdbc:singlestore://%s:%d", conf.host(), conf.port());
-        return DriverManager.getConnection(url, connectionProps);
+
+        try {
+            return DriverManager.getConnection(url, connectionProps);
+        } catch (SQLException e) {
+            if (e.getErrorCode() == 1046 &&
+                    e.getSQLState().equals("3D000") &&
+                    e.getMessage().contains("In order to use shared tier you need to select a database by default")
```

Comment:
the error message may change in the future, can we check only for the error code and state?

Reply:
Yep, we can